cram/sam_header.c

```diff
@@ -812,36 +1014,30 @@ int sam_hdr_rebuild(SAM_hdr *hdr) {
             return -1;
     }
 
-    for (k = kh_begin(hdr->h); k != kh_end(hdr->h); k++) {
-        SAM_hdr_type *t1, *t2;
+    for (i = 0; i < hdr->line_count; i++) {
+        SAM_hdr_type *type;
+        SAM_hdr_tag *tag;
+        char c[2];
 
-        if (!kh_exist(hdr->h, k))
+        if (!hdr->line_order[i].type_data || !strncmp(hdr->line_order[i].type_name, "HD", 2))
             continue;
 
-        if (kh_key(hdr->h, k) == K("HD"))
-            continue;
+        type = hdr->line_order[i].type_data;
+        if (EOF == kputc_('@', &ks))
+            return -1;
 
-        t1 = t2 = kh_val(hdr->h, k);
-        do {
-            SAM_hdr_tag *tag;
-            char c[2];
+        strncpy(c, hdr->line_order[i].type_name, 2);
+        if (EOF == kputsn_(c, 2, &ks))
```

Comment:
I think this could just be `kputsn_(hdr->line_order[i].type_name, 2, &ks)` instead of copying to `c[2]` first.